### ISSUE TITLE
Darktable crashes in Watermark module

### ISSUE THREAD

**Issue by ItaloCunhaBueno** [darktable-org/darktable#4634]
Crashes when trying to add variables in "simple-text" marker, here's the log:

edit: it only happens if you write the variable (even if it's empty like $()) in the beginning of the sentense.

```
this is darktable 3.0.1 reporting an exception:

-------------------

Error occurred on Tuesday, April 7, 2020 at 07:52:46.

darktable.exe caused a Breakpoint at location 00007FFA3B050BB2 in module KERNELBASE.dll.

AddrPC           Params
00007FFA3B050BB2 0000000000000004 0000000000000004 0000000000000004  KERNELBASE.dll!DebugBreak
0000000064A31A2A 0000000000000000 00000000649FE260 0000000000000006  libglib-2.0-0.dll!g_abort
00000000649FD704 000000001F2FE6F8 0000000064ADD440 00000000108DD980  libglib-2.0-0.dll!g_mem_profile
00000000649FE30D 0000000000000000 0000000000000036 000000001F2FE6F0  libglib-2.0-0.dll!g_log_default_handler
00000000649FE65F 000000004CBD0D80 0000000000000004 0000000005DEF710  libglib-2.0-0.dll!g_logv
00000000649FE808 0000000000000000 00007FFA3D129CFC 0000000000000000  libglib-2.0-0.dll!g_log
00000000649FD4ED 0000000100000000 E7FFFFFF00000000 0000000005DEF220  libglib-2.0-0.dll!g_malloc0_n
000000006F28549B 0000000005DEF200 0000000000000008 0000000005DEF2B8  libwatermark.dll!process  [D:/build/darktable/src/iop/watermark.c @ 1024]
00000000637101C2 0000000000000010 0000000005DEF6C9 0000000005DEF6D8  libdarktable.dll! ??   [D:/build/darktable/src/develop/pixelpipe_hb.c @ 2111]
000000006370E776 00000000146B8E10 000000001BF1CAF0 0000000005DEFAA8  libdarktable.dll! ??   [D:/build/darktable/src/develop/pixelpipe_hb.c @ 1281]
0000000063713FCE 00000000019B0000 00007FFA3D23BA17 0000000001930000  libdarktable.dll!dt_dev_pixelpipe_process  [D:/build/darktable/src/develop/pixelpipe_hb.c @ 2650]
00000000636BF546 000000001C1A27D8 00000000031AD3E0 000000001C1A3178  libdarktable.dll!dt_dev_process_image_job  [D:/build/darktable/src/develop/develop.c @ 594]
00000000636941A1 0000000000000002 0000000000000018 000000001C1A3178  libdarktable.dll!dt_dev_process_image_job_run  [D:/build/darktable/src/control/jobs/develop_jobs.c @ 55]
000000006368D11D 0000000004DEDC00 0000000000000000 0000000000000000  libdarktable.dll!dt_control_work_res  [D:/build/darktable/src/control/jobs.c @ 219]
0000000064945092 0000000004DEDA10 0000000000000000 0000000004DEDC00  libwinpthread-1.dll!pthread_create_wrapper
00007FFA3D14B04A 00007FFA3D1A06D0 0000000004DEDC00 0000000000000000  msvcrt.dll!_beginthreadex
00007FFA3D14B11C 0000000000000000 0000000000000000 0000000000000000  msvcrt.dll!_endthreadex
00007FFA3BEF7BD4 0000000000000000 0000000000000000 0000000000000000  KERNEL32.DLL!BaseThreadInitThunk
00007FFA3D26CED1 0000000000000000 0000000000000000 0000000000000000  ntdll.dll!RtlUserThreadStart

darktable.exe	3.0.1.0
ntdll.dll   	10.0.18362.719
KERNEL32.DLL	10.0.18362.752
KERNELBASE.dll	10.0.18362.752
msvcrt.dll  	7.0.18362.1
libglib-2.0-0.dll	2.64.0.0
ADVAPI32.dll	10.0.18362.752
libintl-8.dll	0.19.8.0
sechost.dll 	10.0.18362.693
libdarktable.dll
RPCRT4.dll  	10.0.18362.628
GDI32.dll   	10.0.18362.1
win32u.dll  	10.0.18362.752
gdi32full.dll	10.0.18362.719
ole32.dll   	10.0.18362.693
msvcp_win.dll	10.0.18362.387
combase.dll 	10.0.18362.693
ucrtbase.dll	10.0.18362.387
USER32.dll  	10.0.18362.719
bcryptPrimitives.dll	10.0.18362.295
PSAPI.DLL   	10.0.18362.1
SHELL32.dll 	10.0.18362.719
cfgmgr32.dll	10.0.18362.387
shcore.dll  	10.0.18362.752
libiconv-2.dll	1.16.0.0
windows.storage.dll	10.0.18362.719
profapi.dll 	10.0.18362.693
powrprof.dll	10.0.18362.1
libwinpthread-1.dll	1.0.0.0
UMPDC.dll
libgcc_s_seh-1.dll
shlwapi.dll 	10.0.18362.1
libstdc++-6.dll
kernel.appcore.dll	10.0.18362.1
libcairo-2.dll
cryptsp.dll 	10.0.18362.1
libcurl-4.dll
CRYPT32.dll 	10.0.18362.592
WS2_32.dll  	10.0.18362.387
exchndl.dll 	0.9.2.0
MSASN1.dll  	10.0.18362.1
libexiv2.dll
WLDAP32.dll 	10.0.18362.449
libgdk-3-0.dll	3.24.14.0
IMM32.dll   	10.0.18362.387
libgdk_pixbuf-2.0-0.dll	2.40.0.0
SETUPAPI.dll	10.0.18362.1
bcrypt.dll  	10.0.18362.267
libgmodule-2.0-0.dll	2.64.0.0
libgobject-2.0-0.dll	2.64.0.0
libgphoto2-6.dll
libgphoto2_port-12.dll
libGraphicsMagick-3.dll
libgtk-3-0.dll	3.24.14.0
libIlmImf-2_4.dll
libjpeg-8.dll
libjson-glib-1.0-0.dll
liblcms2-2.dll
lua53.dll
libpango-1.0-0.dll	1.43.0.0
libpangocairo-1.0-0.dll	1.43.0.0
comdlg32.dll	10.0.18362.693
libpng16-16.dll
libpugixml.dll
libsoup-2.4-1.dll
libsqlite3-0.dll
libtiff-5.dll
libxml2-2.dll
zlib1.dll
MSIMG32.dll 	10.0.18362.719
libfontconfig-1.dll
libfreetype-6.dll	2.10.1.0
libpixman-1-0.dll
mgwhelp.dll 	0.9.2.0
VERSION.dll 	10.0.18362.1
libpcre-1.dll
libbrotlidec.dll
libidn2-0.dll
libnghttp2-14.dll
libpsl-5.dll
libssh2-1.dll
libssl-1_1-x64.dll	1.1.1.4
dwmapi.dll  	10.0.18362.267
gdiplus.dll 	10.0.18362.752
WINMM.dll   	10.0.18362.1
libcairo-gobject-2.dll
libepoxy-0.dll
libffi-6.dll
libpangowin32-1.0-0.dll	1.43.0.0
libexif-12.dll
libsystre-0.dll
libbz2-1.dll
libHalf-2_4.dll
libIlmThread-2_4.dll
libthai-0.dll
COMCTL32.dll	6.10.18362.752
libpangoft2-1.0-0.dll	1.43.0.0
libatk-1.0-0.dll	2.34.1.0
WINSPOOL.DRV	10.0.18362.693
libharfbuzz-0.dll
liblzma-5.dll	5.2.4.0
libzstd.dll
dbghelp.dll 	10.0.18362.1
winmmbase.dll	10.0.18362.1
USP10.dll   	10.0.18362.476
libtre-5.dll
libdatrie-1.dll
PROPSYS.dll 	7.0.18362.267
IPHLPAPI.DLL	10.0.18362.1
libgraphite2.dll
OLEAUT32.dll	10.0.18362.693
dbgcore.DLL 	10.0.18362.1
libgomp-1.dll
libgio-2.0-0.dll	2.64.0.0
libopenjp2-7.dll
librsvg-2-2.dll
libsecret-1-0.dll
libexpat-1.dll
libcrypto-1_1-x64.dll	1.1.1.4
libfribidi-0.dll
libltdl-7.dll
libIex-2_4.dll
libImath-2_4.dll
libbrotlicommon.dll
libunistring-2.dll	0.9.10.0
libgcrypt-20.dll	1.8.5.54720
DNSAPI.dll  	10.0.18362.267
NSI.dll     	10.0.18362.449
libcroco-0.6-3.dll
libgpg-error-0.dll	1.37.0.0
CRYPTBASE.DLL	10.0.18362.1
rsaenh.dll  	10.0.18362.1
uxtheme.dll 	10.0.18362.449
clbcatq.dll 	2001.12.10941.16384
mswsock.dll 	10.0.18362.1
MSCTF.dll   	10.0.18362.752
DEVOBJ.dll  	10.0.18362.387
WINTRUST.dll	10.0.18362.387
OpenCL.dll  	2.2.5.0
dxcore.dll  	10.0.18362.1
dxgi.dll    	10.0.18362.693
nvopencl64.dll	26.21.14.4575
nvapi64.dll 	26.21.14.4575
nvcompiler64.dll	26.21.14.4575
winhttp.dll 	10.0.18362.693
IconCodecService.dll	10.0.18362.1
WindowsCodecs.dll	10.0.18362.719
libpixbufloader-png.dll
mscms.dll   	10.0.18362.267
ColorAdapterClient.dll	10.0.18362.267
USERENV.dll 	10.0.18362.387
icm32.dll   	10.0.18362.267
TextInputFramework.dll	10.0.18362.693
CoreUIComponents.dll	10.0.18362.207
CoreMessaging.dll	10.0.18362.1
ntmarta.dll 	10.0.18362.1
wintypes.dll	10.0.18362.693
iertutil.dll	11.0.18362.693
msxml6.dll  	6.30.18362.418
libdarkroom.dll
libknight.dll
liblighttable.dll
libmap.dll
libosmgpsmap-1.0-1.dll
libslideshow.dll
libtethering.dll
libcopy.dll
libexr.dll
libj2k.dll
libjpeg.dll
libpdf.dll
libpfm.dll
libpng.dll
libppm.dll
libtiff.dll
libwebp.dll
libwebp-7.dll
libdisk.dll
libemail.dll
libgallery.dll
liblatex.dll
libpiwigo.dll
libashift.dll
libatrous.dll
libbasecurve.dll
libbasicadj.dll
libbilat.dll
libbilateral.dll
libbloom.dll
libborders.dll
libcacorrect.dll
libchannelmixer.dll
libclahe.dll
libclipping.dll
libcolisa.dll
libcolorbalance.dll
libcolorchecker.dll
libcolorcontrast.dll
libcolorcorrection.dll
libcolorin.dll
libcolorize.dll
libcolormapping.dll
libcolorout.dll
libcolorreconstruct.dll
libcolortransfer.dll
libcolorzones.dll
libdefringe.dll
libdemosaic.dll
libdenoiseprofile.dll
libdither.dll
libequalizer.dll
libexposure.dll
libfilmic.dll
libfilmicrgb.dll
libfinalscale.dll
libflip.dll
libgamma.dll
libglobaltonemap.dll
libgraduatednd.dll
libgrain.dll
libhazeremoval.dll
libhighlights.dll
libhighpass.dll
libhotpixels.dll
libinvert.dll
liblens.dll
liblensfun.dll
liblevels.dll
libliquify.dll
liblowlight.dll
liblowpass.dll
liblut3d.dll
libmask_manager.dll
libmonochrome.dll
libnlmeans.dll
liboverexposed.dll
libprofile_gamma.dll
librawdenoise.dll
librawoverexposed.dll
librawprepare.dll
librelight.dll
libretouch.dll
librgbcurve.dll
librgblevels.dll
librotatepixels.dll
libscalepixels.dll
libshadhi.dll
libsharpen.dll
libsoften.dll
libsplittoning.dll
libspots.dll
libtemperature.dll
libtonecurve.dll
libtoneequal.dll
libtonemap.dll
libvelvia.dll
libvibrance.dll
libvignette.dll
libwatermark.dll
libzonesystem.dll
libbackgroundjobs.dll
libcamera.dll
libcollect.dll
libcolorlabels.dll
libcolorpicker.dll
libcopy_history.dll
libdarktable_label.dll
libduplicate.dll
libexport.dll
libfilmstrip.dll
libfilter.dll
libgeotagging.dll
libglobal_toolbox.dll
libhinter.dll
libhistogram.dll
libhistory.dll
libimage.dll
libimage_infos.dll
libimport.dll
liblighttable_mode.dll
liblive_view.dll
liblocation.dll
libmap_settings.dll
libmasks.dll
libmetadata.dll
libmetadata_view.dll
libmodulegroups.dll
libmodulelist.dll
libmodule_toolbox.dll
libnavigation.dll
libratings.dll
librecentcollect.dll
libselect.dll
libsession.dll
libsnapshots.dll
libstyles.dll
libtagging.dll
libtimeline.dll
libviewswitcher.dll
libview_toolbox.dll
libpixbufloader-svg.dll

Windows 10.0.18363
DrMingw 0.9.2
```

**Comment by phweyland** [comment]
Should be solved by #4483 

this fix protects against null dimensions (when no text) as needed here:

```
    if(dimension.width > dimension.height)
      scale = (iw * roi_out->scale) / dimension.width;
    else
      scale = (ih * roi_out->scale) / dimension.height;
```

**Comment by ItaloCunhaBueno** [comment]
Closed as it was fixed
